test/wdio_functional/page_objects/common/base_page.js

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { testSubjectifySelector } from '../../helpers/helpers';
+import Web from './web';
+
+export default class BasePage extends Web {
+
+  constructor(driver) {
+    super(driver);
+    this.driver = driver;
+  }
+
+  /**
+   * @function nav
+   * @returns {Object} Returns an instance of the class that corresponds to the clicked link.
```

Comment:
Hmmm...maybe we should? Does regular comments serve any purpose for code navigation or anything like that?

Reply:
I agree. I, along with many, use VS Code or something similar. Just helps with ease of use especially for those who aren't as adept in JS. Plus, if we ever migrate to something like Swagger, the docs can be parsed for that module.

Reply:
Sorry, I didn't mean to sound like I was denigrating IDEs, if my comment came across that way. The point I should have made was just that, objectively, if some of us use tools which can leverage jsdoc and some of us aren't, then those who aren't able to leverage jsdoc will probably let the jsdoc become stale or be tempted to remove them (because from that person's perspective, they're a no-op), which will be annoying to the people using jsdoc. No big deal just wanted to mention it.

Reply:
I find jsdoc helpful, especially it can ease the transition to typescript.  Although, to that point, can we just use typescript? I almost went down that rabbit hole earlier but I was not sure if the compiler is set up to capture the test code.

cc @spalger  - do you know if we can use typescript in here, and in the current FTR code?  Or would more work need to be done to support it?